### DISCUSSION TITLE
Trying to do a many to one table need help

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [supabase/supabase#28355] [Questions]
So i want to have an images table where a record say in products could have 5 images and a banner may also have 5 images.

I have done this in the past like so.

id, record_id, record_type
1, 1, product
2, 1, banner

how would i do something like this in supabase and have it automatically cascade. as i dont see the option to have multiple foreign keys on a table

**Comment by GaryAustin1** [comment]
You can have a foreign key with multiple columns using Postgres SQL.  Without checking right now I'm not sure the UI handles an fk with multiple columns.

**Comment by The-Code-Monkey** [comment]
Yeah you can't do it when in the initial table setup screen but when you go
to the edit column screen you can, problem is there is no way of saying
which row would be for which table. E.g.

Row 1, fk 1, type product
Row 2, fk1, type banner

It tries to compare each row to all assigned foreign keys meaning I can't
enter anything into the row as it rejects it on one or more fk.

**Comment by The-Code-Monkey** [comment]
@GaryAustin1 what I actually meant was this. 

https://sqlpad.io/tutorial/mastering-pivot-tables-in-postgresql-for-interviews/#:~:text=In%20PostgreSQL%2C%20a%20pivot%20table,per%20product%20across%20different%20months.

**Reply by The-Code-Monkey** [reply]
Sorry I'm a FE dev learning to do BE stuff so technical terms aren't there yet
